### ISSUE TITLE
Frame transform graph is misleading

### ISSUE THREAD

**Issue by StuartLittlefair** [astropy/astropy#10920]
As noted in #10867 during a discussion with @mhvk and @ayshih the transform graph is a little misleading since it implies, for example, that an ITRS->ITRS' self transform exists. In reality, there is a loopback through ICRS via intermediate frames...

One way of addressing this is to implement a hybrid of #8996 or #7207, so that frame transforms that don't exist are automatically routed through ICRS if possible, or raise an Exception if not.

If we don't like this for some reason, perhaps it would be possible to simply remove these from the graph? Or adding text to the loopback to indicate the frame that is ultimately responsible, like this?

![IMG_798F586AD28E-1](https://user-images.githubusercontent.com/4570807/96899997-7fac3780-1489-11eb-9780-54c2f11ddc5a.jpeg)

pinging @eteq for comments

**Comment by ayshih** [comment]
Assuming that we *don't* automatically self-transform through ICRS:

> If we don't like this for some reason, perhaps it would be possible to simply remove these from the graph?

As a matter of API, the transform graph is an accurate representation.  Each edge between two nodes corresponds to an implemented transformation function between those frames.  I don't like the sound of hiding loopback transformations even when they are defined.

It's important to note that the graph does *not* represent "atomic" transformations.  Each transformation function may internally call other transformation functions in the graph.  This is most common in the loopback transformations, but is also true in other transformations.

> Or adding text to the loopback to indicate the frame that is ultimately responsible, like this?

While this information can be useful for developers to appreciate the underlying implementation, but it changes/increases the objectives of the graph.  Here are some questions:

- Does it help users to see that information?  Put another way, would a user do anything different depending on whether it said ICRS or something else along that edge?
- Should that information be displayed on the same graph as the API, or should there be a separate graph?
- The ITRS loopback is explicitly implemented as a loopback through CIRS, which has its own loopback defined through ICRS.  So, should that edge be annotated as "ICRS" or "CIRS"?  If "ICRS" instead of "CIRS", because that's true extent of the loop, what happens if/when someone implements a CIRS loopback that doesn't go through ICRS?  Do we have to update the annotations on all transformations that might touch that new loopback?
- Since non-loopback transformations can also traverse through intermediate frames, should all the edges be annotated, not just the loopback ones?

**Comment by mkbrewer** [comment]
I don't think that it is possible to implement a CIRS loopback that doesn't go through ICRS when either the observer's location or `obstime` is changed. For solar system bodies this requires a new ephemeris look up. Even for stars, there will be a change in proper motion, parallax, aberration and gravitational light deflection.

**Comment by ayshih** [comment]
> I don't think that it is possible to implement a CIRS loopback that doesn't go through ICRS

Sure, the implementation almost certainly has to go through the ICRS.  But, what if the transformation code uses only ERFA/SOFA routines to go back and forth and doesn't use Astropy's `ICRS` class at all (i.e., it doesn't call any of the other transformations in the graph)?  Does annotating the loopback with "ICRS" add information that is useful or does it add information that is potentially misleading?

Don't get me wrong: I agree that the current graph is misleading.  I'm just really uncertain how to improve it.

**Comment by ayshih** [comment]
For clarity, I recommend that the title of this issue be changed to: "Frame transform graph does not show the use of intermediate frames in transformations"

While we debate more substantial changes, how about at minimum adding a caveat like the following to the narrative for the graph in the docs?
```
This graph shows only the highest-level transformation definitions.
Internally, a transformation may call additional transformations involving
intermediate frames.  For example, the transformation from
`~astropy.coordinates.CIRS` to itself (used for a change in ``obstime``)
internally calls transformations to/from `~astropy.coordinates.ICRS`.
```
